paddle/framework/tensor.cc

```diff
@@ -0,0 +1,32 @@
+#include "paddle/framework/tensor.h"
+
+namespace paddle {
+namespace framework {
+
+int Tensor::Rank() const { return arity(dims_); }
+
+int Tensor::Numel() const { return product(dims_); }
+
+void Tensor::Resize(const DDim& dims) {
```

Comment:
In my design, I don't see that we need `Resize`. Indeed, I followed @qingqing01 's suggestion and removed the ability of the constructor to set the size.